src/pcre2_jit_neon_inc.h

```diff
@@ -82,6 +82,11 @@ POSSIBILITY OF SUCH DAMAGE.
 #  endif
 # endif
 
+#if defined(__GNUC__) || defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeclaration-after-statement"
```

Comment:
Fixing this file to comply with the policy of "declarations at the top" seemed too annoying, so I just disabled the warnings in this file.

We didn't notice about it previously, because this one is only used on Mac (or ARM Linux) and we previously didn't include `-Wdeclaration-after-statement` on those platforms in our CI.

Reply:
clang ALSO defines `__GNUC__`, so there is no need for the second check.

this file is expected to be removed once slits SIMD is also enabled for ARM, and indeed that would solve several known problems with the current implementation.

Reply:
Still a lot of work ahead.